src/pages/index.js

```diff
@@ -0,0 +1,286 @@
+
+  // set card details
+  cardNameEl.textContent = data.name;
+  console.log(data);
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Handle like button click
+  cardLikedBtn.addEventListener("click", (evt) =>
+    handleLike(evt, data._id, data.isLiked)
+  );
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+
+  // Handle delete button click
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  // Handle image preview modals
+  cardImageEl.addEventListener("click", () => {
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function handleClickOverlay(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keyup", handleEscClose);
+  document.addEventListener("click", handleClickOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keyup", handleEscClose);
+  document.removeEventListener("click", handleClickOverlay);
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const activeModal = document.querySelector(".modal_opened");
+    closeModal(activeModal);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDesriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  cardSaveBtn.textContent = "Submitting...";
+
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardslist.prepend(cardElement);
+      closeModal(cardModal);
+      disabledButton(cardSaveBtn, validationConfig);
+      evt.target.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardSaveBtn.textContent = "Save";
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardSaveBtn.textContent = "Save";
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
```

Comment:
Please, change the `confirmation` save button text when the form is being submitted